Updated work-break to break-word over the break-all

garretthyder opened the pull request:
This is to avoid arbitrary word breaks on smaller words Fixes #892
And still fixes issue posed in #875

Commit:
Updated work-break to break-word over the break-all to avoid arbitraty word breaks on smaller words Fixes #892

Mte90 commented:
So we are reverting https://github.com/GlotPress/GlotPress-WP/pull/882/files?
Maybe we need to find a better solution...

toolstack commented:
Doesn't this brings back the old behaviour of breaking the UI on very long strings (without any word breaks in them) like the example in #875?

garretthyder commented:
No reversion here, with #875 the css for word-break didn't exist and then break-all was introduced across the board. This PR switches it from `word-break: break-all` to `word-break: break-word` which still fixes the initial issue in #875 and if you pull up the example and test with inspect you'll see no reversion.
So this change doesn't break the prior issue and fixes this new issue as it now avoid breaking smaller words. This CSS will only break the longer words where it's unavoidable.

Let me know if you need me to create use case examples to proof this

toolstack commented:
@garrett-eclipse it does look to break what we were trying to fix in #875, see below.  A single, very long word will no break properly:

![screenshot-2018-5-2 translations english test glotpress](https://user-images.githubusercontent.com/9053749/39523857-0954af12-4de5-11e8-94d5-6a8d9d841cf0.png)

I'm not sure which is more important though, breaking the UI in some cases or making the translation harder to read in others.

garretthyder commented:
Thanks @toolstack, sorry I was testing using the original string from #875 here;
https://translate.wordpress.org/projects/wp-plugins/duplicate-page/stable-readme/fr/default?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=5633108&filters%5Btranslation_id%5D=53864572

Can you link me to your example there, it appears to wrap on the g but seems to be using more space than the right-hand column. Is that example missing the 45% width on the `table.translations td.original, table.translations td.translation`?

Thank you

toolstack commented:
It's a test project on my local install, here's a PO export of it:

[test-en.po.zip](https://github.com/GlotPress/GlotPress-WP/files/1968265/test-en.po.zip)

The column is pushed out farther than it should because the word never breaks in the middle, it really shows up mostly with URL's.

garretthyder commented:
Alright, sorry it took me some time to spin up a clean instance and reproduce your issue. Turns out it's a Firefox/IE issue as they don't support the `word-break: break-word` css attribute which is why I was seeing it working nicely on Chrome and was confused what was going on. Looking at alternate cross-browser solutions

Commit:
Browser Specific Fallback for word-break on FF and IE

Added CSS selectors for browser support as FF, IE/Edge don't support `word-break: break-word;` so providing to gracefully fallback to break-all on these browsers.

garretthyder commented:
@toolstack can you check my latest commit, I added browser specific fallbacks for FF/IE/Edge as they don't support `word-break: break-word;` so they'll become break-all
SO sadly these browsers will break the smaller words too and all this was just to get Chrome/Safari/Opera and webkit browsers working nicely.

toolstack commented:
@garrett-eclipse Looks good to me.

garretthyder commented:
Thanks @toolstack 
But I think this might need to be closed.

@ocean90 had implemented some changes to translate.wordpress.org for this, see comments on #892 for tests.

I'll defer to Dominik on the path forward for this functionality as I believe his implementation doesn't require browser specific fallbacks. However I believe it was committed directly to meta trac so may need to be ported into this project.

Feel free to reopen this PR if there's any use.
